### ISSUE TITLE
P_PreAssemblerDagcon hgapCorrection jobs fail.

### ISSUE THREAD

**Issue by tolot27** [PacificBiosciences/SMRT-Analysis#138]
There seem to be a bug in the hgapCorrection task of the P_PreAssemberDragcon module (v2.1.124285 of SMRT-Analysis 2.3.0 patch 0):

All chunks, except 1, of the hgapCorrection tasks fail because they try to access the input file for chunk 1. I. e. hgapCorrection_004of006.log:

```log
Validating existence of Input Files
Successfully found /storage/fungalstore/other/pacbio/jobs/016/016587/.seeds.m4.filter
Successfully found /storage/fungalstore/other/pacbio/jobs/016/016587/data/filtered_subreads.fasta
Successfully found /storage/fungalstore/other/pacbio/jobs/016/016587/seeds.chunk004of006.m4
Successfully found /storage/fungalstore/other/pacbio/jobs/016/016587/seeds.m4.fofn
Successfully validated input files
Generating pre-alignments
m4topre.py /storage/fungalstore/other/pacbio/jobs/016/016587/seeds.chunk004of006.m4 /storage/fungalstore/other/pacbio/jobs/016/016587/seeds.m4.fofn /storage/fungalstore/other/pacbio/jobs/016/016587/data/filtered_subreads.fasta 24 > /home/biomax/soft/smrtanalysis/tmpdir/tmp.wOprTKPooX/aln.12969.pre
Traceback (most recent call last):
  File "/home/biomax/soft/smrtanalysis/install/smrtanalysis_2.3.0.140936/analysis/bin/m4topre.py", line 208, in <module>
    sys.exit(main())
  File "/home/biomax/soft/smrtanalysis/install/smrtanalysis_2.3.0.140936/analysis/bin/m4topre.py", line 149, in main
    m4h = open(m4f)
IOError: [Errno 2] No such file or directory: '/storage/fungalstore/other/pacbio/jobs/016/016587/seeds.chunk001of006.m4'
Task 1 completed at Fri Oct 17 16:54:33 CEST 2014
Finished on Fri Oct 17 14:54:33 UTC 2014
Task hgapCorrection_004of006 with nproc 44 with exit code 0.
# Writing stdout and stderr from Popen:
Your job 944259 ("Phga016587") has been submitted
Job 944259 exited with exit code 0.
```

Look at the IOError!

Also, the exit code (0 in this case of an error) is wrong and hence, the task cannot be claimed to be completed.
Interestingly, no jobs are running but the workflow is not stopped and the SMRT Portal still monitors the job as a running job.

I've verified that all files seeds.chunk00?of006.m4 exist. That's the case. They have all about the same size (250 MB). Maybe the hgapCorrection tasks are started to earlier, before all files 'seeds.chunk00?of006.m4' are created?

BTW: The same job worked quite well with SMRTAnalysis 2.2.0 + latest patch.

**Comment by pb-jlandolin** [comment]
Ok, maybe it is a filesystem lag issue.  I know we only use NFS, and our code doesn't quite work with other types such as GlusterFS as the file existence checks fail.  I'll pass your fix on to the developers so they at least know a way to get around it.

**Comment by tolot27** [comment]
It is okay to close this issue. I made some further investigations and found out that it is maybe related to the switch of our cluster software. We switched from Sun Grid Engine (SGE) to Univa Grid Engine (UGE). That is basically a fork of the SGE with some bugfixes and enhancements. We discovered two issues with UGE. One is related to Kerberos authentication against Active Directory (where our user accounts are stored) and the other one related to secondary group id resolution.

We did not resolve the reported issue but my workaround posted above works still quite nicely.
